src/parsing.jl

```diff
@@ -368,6 +422,7 @@ end
 ###
 
 function parse_row!(rec::R, bytes, pos, len, options) where {I, R <: MultiTerminalDCLines{I}}
+    pos = checkdelim!(bytes, pos, len, options)
```

Comment:
@quinnj how does CSV.jl handle leading whitespace at the start of a row (when whitespace is the delim)? Does CSV.jl manage to avoid calling `checkdelim!` in every `parserow` call somehow?

Reply:
In `Parsers.checkcmtemptylines`, we check if a new row starts with a comment character or is "empty" (i.e. is followed immediately by another newline). We call that anywhere right after we parse a newline.

Otherwise, individual type parsers have their own logic for skipping whitespace, so the "first cell" of the row would be able to skip leading whitespace for the row.

If the delim is whitespace, however, then by nature it's significant, right? So we would treat it as empty cells. (i.e. consecutive delimiters like `,,` mean an empty cell.

UNLESS we're talking `ignorerepeated=true`, in which case, yes, we'll parse the newline _and any following delimiters_ until we reach a non delimiter which should be the start of our first cell.

Reply:
yeah, sorry should have been clearer, our case here is `delim=' '` and `ignorerepeated=true`

Basically we have space-seperated values, which can be seperated by an arbitrary number of spaces, _and this first value might also have arbitrary number of leading spaces_ 

e.g. data with 5 columns can look like:
```
    8 'ABC     ' 138.00 1      .000     
   19 'ABCDEFGH'  69.00 1      .000 
```

In CSV, i could see _for the very first row_ we handle this case in `Context` here:
https://github.com/JuliaData/CSV.jl/blob/3ebd2c9d3baec32512a662605cf0fc378fdf1644/src/context.jl#L382-L389
```jl
        # step 4a: if we're ignoring repeated delimiters, then we ignore any
        # that start a row, so we need to check if we need to adjust our headerpos/datapos
        if ignorerepeated
            if headerpos > 0
                headerpos = Parsers.checkdelim!(buf, headerpos, len, options)
            end
            datapos = Parsers.checkdelim!(buf, datapos, len, options)
        end
```
but i couldn't see how we handle this in all other lines

turns out `xparse` handles this itself, and the problem in PowerFlowData.jl was that out `next_line` function which we use to skip lines between sections (and which is a simplified version of `Parsers.checkcmtemptylines`) wasn't handling repeated delimiters at the start of the next line

thanks for the pointers -- changes needed here are much simpler now!